include/seqan3/search/algorithm/detail/search_trivial.hpp

```diff
@@ -101,24 +119,33 @@ inline bool search_trivial(cursor_t cur, query_t & query, typename cursor_t::siz
                     // Match (when error_left.substitution == 0)
                     if (error_left.substitution == 0 && cur.last_rank() == to_rank(query[query_pos]))
                     {
-                        if (search_trivial<abort_on_hit>(cur, query, query_pos + 1, error_left, delegate) &&
+                        if (search_trivial<abort_on_hit>(cur, query, query_pos + 1, error_left,
+                                                         error_type::MATCHMM, delegate) &&
                             abort_on_hit)
                         {
                             return true;
                         }
                     }
 
-                    // Deletions at the end of the sequence are not allowed. This cannot happen: when the algorithm
+                    // Deletions at the end of the sequence are not allowed. This cannot happen: When the algorithm
                     // arrives here, it cannot be at the end of the query and since deletions do not touch the query
                     // (i.e. increase query_pos) it won't be at the end of the query after the deletion.
-                    if (error_left.deletion > 0)
+                    // Do not allow deletions after an insertion.
+                    if ((prev_error != error_type::INSERTION || error_left.substitution == 0) &&
+                        error_left.deletion > 0)
                     {
                         search_param error_left2{error_left};
                         error_left2.total--;
                         error_left2.deletion--;
-
-                        if (search_trivial<abort_on_hit>(cur, query, query_pos, error_left2, delegate) && abort_on_hit)
-                            return true;
+                        // Only search for characters different from the corresponding query character.
+                        // (Same character is covered by a match.)
+                        if (cur.last_rank() != to_rank(query[query_pos]))
```

Comment:
```suggestion
                        if (cur.last_rank() != seqan3::to_rank(query[query_pos]))
```

Reply:
Always call CPOs with a qualified name since the unqualified lookup is handled by the implementation of the CPO